dev-scheme/racket/racket-8.2.ebuild

```diff
@@ -0,0 +1,172 @@
+		xdg_desktop_database_update
+		xdg_icon_cache_update
+	fi
+}
+
+src_prepare() {
+	# Prepare environment
+	unset PLTUSERHOME
+	xdg_environment_reset
+
+	default
+
+	# Remove bundled libffi
+	rm -r ./bc/foreign/libffi || die "failed to remove bundled libffi"
+}
+
+src_configure() {
+	# According to vapier, we should use the bundled libtool
+	#   such that we don't preclude cross-compile.
+	#   Thus don't use --enable-lt=/usr/bin/libtool
+	# --enable-bc builds Racket w/o chez backend
+	local myconf=(
+		--disable-libs
+		--disable-strip
+		--docdir="${EPREFIX}/usr/share/doc/${PF}"
+		--enable-float
+		--enable-foreign
+		--enable-libffi
+		--enable-shared
+		$(usex chez "--enable-cs --enable-csonly" "--enable-bc --enable-bconly")
+		$(use_enable X gracket)
+		$(use_enable doc docs)
+		$(use_enable jit)
+		$(use_enable places)
+		$(use_enable futures)
+		$(use_enable threads pthread)
+	)
+	econf "${myconf[@]}"
+}
+
+src_compile() {
+	# When the JIT is enabled, a few binaries need to be pax-marked
+	# on hardened systems (bug 613634). The trick is to pax-mark
+	# them before they're used later in the build system. The
+	# following order for racketcgc and racket3m was determined by
+	# digging through the Makefile in src/racket to find out which
+	# targets would build those binaries but not use them.
+	if use jit && ! use chez; then
+		pushd ./bc  || die
+		emake cgc-core
+		pax-mark m .libs/racketcgc
+
+		pushd ./gc2 || die
+		emake all
+		popd || die
+
+		pax-mark m .libs/racket3m
+		popd || die
+	fi
+
+	default
+}
+
+src_install() {
+	default
+
+	# When the JIT is enabled, the final binaries need to be pax-marked, too,
+	# if you want to actually use them. The src_compile marking get lost
+	# somewhere in the install process.
+	if use jit; then
+		local f
+		for f in mred mzscheme racket; do
+			pax-mark m "${D}/usr/bin/${f}"
+		done
+
+		pax-mark m "${D}/usr/$(get_libdir)/racket/starter"
+		use X && pax-mark m "${D}/usr/$(get_libdir)/racket/gracket"
+	fi
+
+	# raco needs decompressed files for packages doc installation bug 662424
+	if use doc; then
+		docompress -x "/usr/share/doc/${PF}"
+	fi
+
+	# Remove unneeded "*.a" and "*.la"
+	find "${ED}" \( -name "*.a" -o -name "*.la" \) -delete || die
+
+	# Create missing desktop files and icon
+	if use X; then
+		newicon "${D}/usr/share/racket/drracket-exe-icon.png" "racket.png"
+		make_desktop_entry "gracket" "GRacket" "racket" "Development;Education;"
+		make_desktop_entry "plt-games" "PLT Games" "racket" "Education;Game;"
+	fi
+}
+
+pkg_preinst() {
+	# pkg database files
+	local pkgdb=(
+		/usr/share/racket/info-cache.rktd
+		/usr/share/racket/links.rktd
+		/usr/share/racket/pkgs/pkgs.rktd
+	)
+	# If we are merging the same version (not revision!)
+	# check if pkg database files exist and do not overwrite them
+	if has_version "~${CATEGORY}/${PN}-${PV}"; then
+		echo "We are installing the same version: ${PV}"
+		local rktd
+		for rktd in "${pkgdb[@]}"; do
+			if [ -f "${EROOT}/${rktd}" ]; then
```

Comment:
Bash test [please](https://devmanual.gentoo.org/tools-reference/bash/#single-versus-double-brackets-in-bash)!